src/editions/SingleBatchEditionFactory.sol

```diff
@@ -74,15 +101,41 @@ contract SingleBatchEditionCreator {
             }
         }
 
-        emit CreatedBatchEdition(uint256(salt), creator, address(edition), tags);
+        edition.mintBatch(pointer);
+
+        emit CreatedBatchEdition(uint256(salt), creator, address(edition), data.tags);
 
-        configureEdition(edition, signedAttestation, externalUrl, creatorName);
+        configureEdition(edition, signedAttestation, data.externalUrl, data.creatorName);
 
         // and finally transfer ownership of the configured contract to the actual creator
         IOwnable(address(edition)).transferOwnership(creator);
     }
 
-    function getEditionAtId(uint256 editionId) external view returns (ISingleBatchEdition) {
+    /// Creates and mint a new SingleBatchEdition contract with a deterministic address
+    /// @dev we expect the signed attestation's context to correspond to this contract's address
+    /// @dev we expect the signed attestation's beneficiary to be the edition's creator
+    /// @param packedRecipients an abi.encodePacked() array of recipient addresses for the batch mint
+    /// @param signedAttestation a signed message from Showtime authorizing this action on behalf of the edition creator
+    /// @return edition the address of the created edition
+    function createEdition(
+        EditionData calldata data,
+        bytes calldata packedRecipients,
+        SignedAttestation calldata signedAttestation
+    ) external returns (ISingleBatchEdition edition) {
+        if (packedRecipients.length == 0 || packedRecipients.length % 20 != 0) {
```

Comment:
```
packedRecipients.length % 20 != 0
```
Is this checking the amount of addresses? So this would technically be the minimum threshold we discussed on our call?

Reply:
My guess is no since I also read the tests 😅 
So then this is checking that each wallet address is 20 bytes long?

Reply:
yes it's checking that the recipients looks like a valid list of addresses (i.e. a multiple of 20 bytes)

Reply:
TIL addresses are 20 bytes long!